storage/tests/collision.rs

```diff
@@ -0,0 +1,32 @@
+mod common;
+
+use {
+    eth_trie::DB,
+    hex_literal::hex,
+    moved::primitives::B256,
+    moved_storage::{RocksEthTrieDb, ROOT_KEY},
+};
+
+#[test]
+fn test_column_families_do_not_collide() {
+    let rocks = common::create_db();
+    let db = RocksEthTrieDb::new(&rocks);
+
+    let expected_root = B256::new(hex!(
+        "50596cee391a497683672d9396379f56cd8e96476b844557933f48039c483a81"
+    ));
+    db.put_root(expected_root).unwrap();
+
+    let expected_value = hex!("feef");
+    db.insert(ROOT_KEY.as_bytes(), expected_value.as_slice().to_vec())
```

Comment:
I think we should leave  a comment saying, using the same key (`ROOT_KEY`) for root and tree element should not cause any collision.